### ISSUE TITLE
Proxy remains enabled after updating config

### ISSUE THREAD

**Issue by chimericdream** [tmatilai/vagrant-proxyconf#87]
I regularly use my VM from behind a proxy at work an without a proxy at home. Going from home to work performs as expected. I explicitly set the `config.proxy.enabled` to `true`, and the plugin reconfigures the machine to use the given proxy settings. However, when going from work to home, the VM will not provision, even with `config.proxy.enabled` set to `false`.

Configuration at work:

```
  if Vagrant.has_plugin?("vagrant-proxyconf")
    config.proxy.http     = "http://SOME_IP"
    config.proxy.https    = "http://SOME_IP"
    config.proxy.no_proxy = "localhost,127.0.0.1"
    config.proxy.enabled  = true
  end
```

Configuration at home:

```
  if Vagrant.has_plugin?("vagrant-proxyconf")
    config.proxy.http     = "http://SOME_IP"
    config.proxy.https    = "http://SOME_IP"
    config.proxy.no_proxy = "localhost,127.0.0.1"
    config.proxy.enabled  = false
  end
```

I would expect the `enabled` setting to force the plugin to reconfigure the proxy.

**Comment by codylane** [comment]
Hello, I believe I've addressed this concern in my local test environment and will be putting in a PR to address this.  FWIW, this is a bug and the original poster is correct that no matter what the README says to do it will not disable the proxy once it has been configured it just has logic to "skip" configuring the proxy if the plugin is disabled.  

This took me a while to understand but I believe I've got that logic sorted out and I will address each change with tests.  Look for a PR in the next few days.  I'll also link this issue to the PR.
